test/support/requests/plugins/stream_bidi.rb

```diff
@@ -74,6 +74,35 @@ def test_plugin_stream_bidi_buffer_data_from_separate_thread
           assert chunks.size == 5, "all the lines should have been yielded"
         end
       end
+
+      def test_plugin_stream_bidi_buffer_thread_safety
```

Comment:
I believe this test should exercise the public API. For example, you mentioned that you had issues using this in a puma server. was it getting the response and then streaming from a different thread? was it something with the request? I'd rather have a test closer to the failed use case, rather than just verifying whether the bidi-buffer really locks around buffer ops.